content-tools/update-cdids/main.go

```diff
@@ -0,0 +1,443 @@
+	ctx, err := authenticate(ctx, httpClient, environment, username, password)
+	if err != nil {
+		_ = fmt.Errorf("error occurred while authenticating. Stopping the script. error: %s", err.Error())
+		return
+	}
+
+	collectionID, err := createCollection(ctx, httpClient, getCollectionName(), environment)
+	if err != nil {
+		fmt.Errorf("error occurred while creating collection. Stopping the script. error: %s", err.Error())
+		return
+	}
+	log.Event(ctx, fmt.Sprintf("created collection: %s", collectionID), log.INFO)
+
+	for _, pair := range cdIDPairs {
+		pair.Print(ctx, "Processing ")
+
+		oldCdIDLocation, err := searchOldCdID(ctx, httpClient, environment, pair.oldCdID)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while fetching cdID location")
+			continue
+		}
+
+		cdIDData, err := fetchDataForCDID(ctx, httpClient, environment, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while fetching cdID data.")
+			continue
+		}
+
+		err = addCDIDToCollection(ctx, httpClient, environment, collectionID, oldCdIDLocation, cdIDData)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while adding CDID to collection")
+			continue
+		}
+
+		err = approveCDID(ctx, httpClient, environment, collectionID, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while approving cdid in the collection")
+			continue
+		}
+
+		pair.Print(ctx, "Completed Processing ")
+	}
+
+	log.Event(ctx, "successfully updated all documents.", log.INFO)
+}
+
+// 	curl -X POST --header "X-Florence-Token:$ZEBEDEE_TOKEN"
+//	http://localhost:8082/review/{collection-id}?uri={path-to-content}/data.json&recursive=false
+func approveCDID(ctx context.Context, client *http.Client, environment string, collectionID string, cdIDLocation string) error {
+	cdIDURI := strings.Replace(cdIDLocation, environment, "", -1)
+
+	pageReviewURL := fmt.Sprintf("%s/review/%s?uri=%s/data.json&recursive=false", environment, collectionID, cdIDURI)
+	req, err := http.NewRequestWithContext(ctx, "POST", pageReviewURL, nil)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to approve CDID. Error: %v", err)
+		log.Error(errMessage)
+		return errMessage
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set(FlorenceToken, ctx.Value(ZebedeeToken).(string))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to approve CDID. Error: %v", err)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	if resp.StatusCode > 400 {
+		errMessage := fmt.Errorf("failed to approve CDID. Error In API: %v. Status Code: %s", err, resp.Status)
+		log.Error(errMessage)
+
+		return errMessage
+	}
+	return nil
+}
+
+// POST http://localhost:8081/zebedee/content
+//		/test-4c153d5de19c33be41b772b9d6c27dfc917bda71f2d8925e6b95e17da3a7d8cf
+//		?uri=inflationandpriceindices/timeseries/mb55/mm22/data.json
+//		&overwriteExisting=true
+
```

Comment:
⚠️ I think you may possibly need to check if the content being added is already in another collection. I am not sure this endpoint will do that as part of handling this request. I think this is unlikely but we need to check because we could get into a mess with the data - I assume this will be used on prod/live data at some point so we **have** to be cautious and check this.

Simple test you can do - create another collection (A), update a page and add a page to the collection A. Run this script and try to add the same page to collection (B). If you don't get an error then we need to add a manual step to do that check.

I'm not sure if there is a stand alone endpoint to do that already - if the test above fails Carl or I can help you work out a way to do the check.